Yoast/yoast-acf-analysis
# CS: rename (test) classses and rename PHP files

## Summary

This PR can be summarized in the following changelog entry:
* _N/A_

## Relevant technical choices:
* No functional changes.
* Code style compliance.

As discussed with @moorscode.

1. Renames test classes to use underscores to separate words
2. Adjusts the file names for class files to be in line with the rules as laid out in the `Yoast.Files.FileName` sniff.
    Ref: Yoast/yoastcs#48
    * Remove `class-` prefixes from the plugin file names.
    * Remove `yoast-acf-analysis` prefixes from the plugin file names.
    * Make sure that the file names accurately reflect the class names (excluding the prefixes).
3. **IMPORTANT** I noticed when committing the file renames, that the file `mode` for a number of these files has also been changed from `755` to `644`. This change was unintentional, however, does look like a more sensible file mode to me.
Please let me know if this is undesired/if there was any special reason why these files had mode `755`.


Where necessary, adjusted any `include`/`require`s to point to the new file names.

- [ ] **Note:** The classmap autoload files for Composer will need to be regenerate!


## Test instructions

This PR can be tested by following these steps:

* * See if the plugin still loads as expected, functions as expected, unit tests still run as expected etc.
